R/build.gradle

```diff
@@ -0,0 +1,96 @@
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+    id 'io.deephaven.project.register'
+    id 'io.deephaven.deephaven-in-docker'
+}
+
+evaluationDependsOn Docker.registryProject('r-client-base')
+
+configurations {
+    cpp {}
+}
+
+dependencies {
+  cpp project(':cpp-client')
+}
+
+def prefix = '/opt/deephaven'
+
+// start a grpc-api server
+String randomSuffix = UUID.randomUUID().toString();
+deephavenDocker {
+    envVars.set([
+            'START_OPTS':'-Xmx512m -DAuthHandlers=io.deephaven.auth.AnonymousAuthenticationHandler'
+    ])
+    containerName.set "dh-server-for-r-${randomSuffix}"
+    networkName.set "r-test-network-${randomSuffix}"
+}
+
+def testRClient = Docker.registerDockerTask(project, 'testRClient') {
+    // Only tested on x86-64, and we only build dependencies for x86-64
+    platform = 'linux/amd64'
+
+    copyIn {
+        from(layout.projectDirectory) {
+            include 'r-tests.sh'
+            include 'rdeephaven/DESCRIPTION'
+            include 'rdeephaven/LICENSE'
+            include 'rdeephaven/NAMESPACE'
+            include 'rdeephaven/README.md'
+            include 'rdeephaven/inst/**'
+            include 'rdeephaven/man/**'
+            include 'rdeephaven/etc/**'
+            include 'rdeephaven/R/**'
+            include 'rdeephaven/src/*.cpp'
+            include 'rdeephaven/src/Makevars'
+        }
+    }
+    copyOut {
+        into layout.buildDirectory.dir('test-results')
+    }
+    dockerfile {
+        from('deephaven/cpp-client:local-build')
+        //
+        // Build and install client.
+        //
+        runCommand("""mkdir -p \\
+                        /out \\
+                        ${prefix}/log \\
+                        ${prefix}/bin/rdeephaven \\
+                        ${prefix}/src/rdeephaven/{inst,man,etc,src,R,bin}
+                   """)
+        copyFile('rdeephaven/DESCRIPTION',  "${prefix}/src/rdeephaven/")
+        copyFile('rdeephaven/LICENSE',  "${prefix}/src/rdeephaven/")
+        copyFile('rdeephaven/NAMESPACE',  "${prefix}/src/rdeephaven/")
+        copyFile('rdeephaven/README.md',  "${prefix}/src/rdeephaven/")
+        copyFile('rdeephaven/inst/',  "${prefix}/src/rdeephaven/inst/")
+        copyFile('rdeephaven/man/', "${prefix}/src/rdeephaven/man/")
+        copyFile('rdeephaven/etc/', "${prefix}/src/rdeephaven/etc/")
+        copyFile('rdeephaven/R/', "${prefix}/src/rdeephaven/R/")
+        copyFile('rdeephaven/src/*.cpp', "${prefix}/src/rdeephaven/src/")
+        copyFile('rdeephaven/src/Makevars', "${prefix}/src/rdeephaven/src/")
+        copyFile('r-tests.sh', "${prefix}/bin/rdeephaven")
+        runCommand("PREFIX=\"" + prefix + "\"; \\\n" +
+                  '''set -eux ; \
+                     cd "${PREFIX}/src/rdeephaven"; \
+                     . "${PREFIX}/env.sh"; \
+                     echo 'install.packages(".", repos=NULL, type="source")' | \
```

Comment:
There might be reason to want to do this via
```
R CMD build rdeephaven
R CMD check rdeephaven*.tar.gz
R CMD INSTALL rdeephaven*.tar.gz
```
from terminal in the `${PREFIX}/src` directory, rather than getting into an R console and running `install.packages` directly.
`R CMD check` runs a laundry list of [checks](https://kbroman.org/pkg_primer/pages/check.html) that, if we ever make it on CRAN, CRAN will run as well. This gives us an extremely verbose and detailed look at what are package is doing, if all the documentation is as it should be, if the package structure is as it should be, etc etc.

I think it would not be the case that we want to call CI "failed" if `R CMD check` complains about something but all of the unit tests pass. Maybe we would, or maybe there's an intermediate stage between passing and failing CI that says "tests don't fail, but you should look at the output from R CMD check and see what's up".

Reply:
We discussed on slack and agreed that for automated CI checks, we want to be doing the same thing that we are asking the customer to do in the README regarding the manual steps they would run (so that we continue to check that what we are asking users to do to build works).